Objects/abstract.c

```diff
@@ -495,6 +494,47 @@ _Py_add_one_to_index_C(int nd, Py_ssize_t *index, const Py_ssize_t *shape)
     }
 }
 
+int
+PyBuffer_SizeFromFormat(const char *format)
+{
+    PyObject *structmodule;
+    PyObject *Struct = NULL;
+    PyObject *structobj = NULL;
+    PyObject *fmt = NULL;
+    int itemsize;
+
+    structmodule = PyImport_ImportModule("struct");
+    if (structmodule == NULL)
+        goto error;
+
+    Struct = PyObject_GetAttrString(structmodule, "calcsize");
+    Py_DECREF(structmodule);
+    if (Struct == NULL)
+        goto error;
+
+    fmt = PyBytes_FromString(format);
+    if (fmt == NULL)
+        goto error;
+
+    structobj = PyObject_CallFunctionObjArgs(Struct, fmt, NULL);
+    if (structobj == NULL)
+        goto error;
+
+    itemsize = (intptr_t)structobj;
```

Comment:
I don't think that it works. structobj is a Python object. You should use PyLong_AsSsize_t() to get it as Py_ssize_t.